src/config.rs

```diff
@@ -438,48 +451,51 @@ impl Config {
             display,
             updates,
             directories,
+            file_path: config_file_path,
         })
     }
 
-    pub fn load(enable_styles: bool) -> Result<Self> {
-        debug!("Loading config");
+    /// Load and read the config file from the given path into
+    /// a [Config] and return it.
+    ///
+    /// path: The path to the config file.
+    pub fn load(path: &Path) -> Result<Self> {
+        let raw_config = RawConfig::load(File::open(path)?)?;
+
+        let config = Self::from_raw(
+            raw_config,
+            PathWithSource {
+                path: path.into(),
+                source: PathSource::Cli,
+            },
+        )
+        .context("Could not process raw config")?;
 
+        Ok(config)
+    }
+
+    /// Load and read the config file from the default path into
+    /// a [Config] and return it.
+    pub fn load_default_path() -> Result<Self> {
         // Determine path
-        let (config_file_path, _) = get_config_path().context("Could not determine config path")?;
-
-        // Load raw config
-        let raw_config: RawConfig = if config_file_path.exists() && config_file_path.is_file() {
-            let mut config_file = fs::File::open(&config_file_path).with_context(|| {
-                format!("Failed to open config file path at {:?}", &config_file_path)
-            })?;
-            let mut contents = String::new();
-            config_file.read_to_string(&mut contents).with_context(|| {
-                format!("Failed to read from config file at {:?}", &config_file_path)
-            })?;
-            toml::from_str(&contents).with_context(|| {
-                format!("Failed to parse TOML config file at {config_file_path:?}")
-            })?
-        } else {
-            RawConfig::new()
+        let config_file_path =
+            get_default_config_path().context("Could not determine config path")?;
+
+        let raw_config = match File::open(config_file_path.path()) {
+            Ok(file) => RawConfig::load(file)?,
+            Err(e) => {
+                if e.kind().eq(&ErrorKind::NotFound) {
+                    RawConfig::default()
+                } else {
+                    bail!(
+                        "Failed to open config file at {}",
+                        config_file_path.path().display()
+                    );
+                }
+            }
```

Comment:
I think we can make this slightly nicer as follows:
```suggestion
            Ok(file) => RawConfig::load(file)?,
            Err(e) if e.kind() == ErrorKind::NotFound => RawConfig::default(),
            Err(e) => {
                return Err(e).context(format!(
                    "Failed to open config file at {}",
                    config_file_path.path().display()
                ));
            }
```

This way we also keep the information from the error in the case we return here

Reply:
That makes sense!
I committed your suggestion from here :)